modules/spring-config/src/test/java/org/mule/runtime/config/spring/SpringRegistryTestCase.java

```diff
@@ -71,4 +90,65 @@ private void createSpringRegistryWithParentContext() {
     parentApplicationContext = new StaticApplicationContext();
     springRegistry = new SpringRegistry(REGISTERY_ID, applicationContext, parentApplicationContext, null);
   }
+
+  @Test
+  public void registerObjectAsynchronously() throws MuleException, InterruptedException, ExecutionException {
+    final MuleContext muleContext = mock(MuleContext.class);
+    springRegistry = new SpringRegistry(new StaticApplicationContext(), muleContext);
+    springRegistry.initialise();
+    springRegistry.getLifecycleManager().fireLifecycle(Startable.PHASE_NAME);
+
+    final Startable asyncStartableBean = mock(Startable.class, withSettings().extraInterfaces(Disposable.class));
+
+    final Latch startingLatch = new Latch();
+
+    doAnswer(invocation -> {
+      startingLatch.countDown();
+      sleep(1000);
```

Comment:
"waiting for some time" is the definition of flaky. In the bug you're trying to fix, why was this object being wrongfully disposed? what was triggering that disposal? You can add a second latch which is fired by that condition and THEN you verify that it was never disposed.